h11/_receivebuffer.py

```diff
@@ -79,21 +90,29 @@ def maybe_extract_at_most(self, count):
         self._start += len(out)
         return out
 
-    def maybe_extract_until_next(self, needle):
+    def maybe_extract_until_next(self, needle_regex, max_needle_length):
         # Returns extracted bytes on success (advancing offset), or None on
         # failure
-        if self._looked_for == needle:
-            search_start = max(self._start, self._looked_at - len(needle) + 1)
+        if self._looked_for_regex == needle_regex:
+            looked_at = max(self._start, self._looked_at - max_needle_length)
         else:
-            search_start = self._start
-        offset = self._data.find(needle, search_start)
-        if offset == -1:
+            looked_at = self._start
+            self._looked_for_regex = needle_regex
+
+        delimiter_match = next(
+            self._looked_for_regex.finditer(self._data, looked_at), None
+        )
```

Comment:
Could you help explain why this is `next(finditer(...))` rather than `search(...)` here?
What am I missing?

Reply:
Thank you, you are absolutely right, I missed `Pattern::search` using `Pattern::finditer`. I'm to fix it using `search` as it's more clear